Use Ceph monitoring (not RGW) network for Dashboard

fultonj opened the pull request:
Use Ceph monitoring (not RGW) network for Dashboard

Use the existing `cifmw_cephadm_monitoring_network`
parameter to configure the Ceph dashboard instead of
the `cifmw_cephadm_rgw_network` parameter. This was
not detected earlier because the Ceph playbook set them
both to the Ceph public_network, but we recently added
a feature to run RGW on a separate network [1].
Also, add retries/delay to the dashboard HTTP check.

[1] https://github.com/openstack-k8s-operators/ci-framework/pull/2301

fultonj commented:
The `cifmw_cephadm_monitoring_network` is set here:

  https://github.com/openstack-k8s-operators/ci-framework/blob/main/playbooks/ceph.yml#L435

fmount commented on roles/cifmw_cephadm/tasks/dashboard/configure_dashboard_backends.yml:
```diff
@@ -34,5 +34,5 @@
     {{ cifmw_cephadm_ceph_cli }} config set \
     mgr mgr/dashboard/{{ current_mgr }}/server_addr \
-    {{ hostvars[item][all_addresses] | ansible.utils.ipaddr(cifmw_cephadm_rgw_network) | first }}
+    {{ hostvars[item][all_addresses] | ansible.utils.ipaddr(cifmw_cephadm_monitoring_network) | first }}
```
+1 thanks

katarimanojk replied:
+1  considering the changes to `cifmw_cephadm_rgw_network` in #2301, where RGW can use a saperate nework. 
this is very much needed here.

fmount commented on roles/cifmw_cephadm/tasks/dashboard/validation.yml:
```diff
@@ -1,15 +1,47 @@
 - name: Validate connection to dashboard service
   ansible.builtin.get_url:
-    url: "http://{{ grafana_server_addr }}:{{ cifmw_cephadm_dashboard_port }}"
+    url: "{{ cifmw_cephadm_urischeme_dashboard | default('http') }}://{{ grafana_server_addr }}:{{ cifmw_cephadm_dashboard_port }}"
```
ah! that makes sense and explain the connection refused!

fmount reviewed (approved):
/lgtm

fmount commented on roles/cifmw_cephadm/tasks/dashboard/validation.yml:
```diff
@@ -1,15 +1,47 @@
+  ansible.builtin.set_fact:
+    cifmw_cephadm_urischeme_dashboard: "https"
+  when:
+    - cifmw_cephadm_dashboard_crt | length > 0
```
The job is currently failing because `cifmw_cephadm_dashboard_crt' is undefined`.
I suppose we want to put an additional `cifmw_cephadm_dashboard_crt is defined` in the condition

fmount reviewed (changes requested):
cifmw_cephadm_dashboard_crt' is undefined is the current failure in validation.yaml.
I assume we can change the urischeme and set it to https only when that variable is defined/passed as input.

fultonj commented:
> cifmw_cephadm_dashboard_crt' is undefined is the current failure in validation.yaml. I assume we can change the urischeme and set it to https only when that variable is defined/passed as input [1].
> 
> [1] https://logserver.rdoproject.org/95/2395/b0951c95ffd67b13d9c69923d97211b7bd1294fd/github-check/podified-multinode-hci-deployment-crc/cc39af8/job-output.txt

OK, thanks. I updated accordingly.

```yaml
  when:
    - cifmw_cephadm_dashboard_crt | default("") | length > 0
    - cifmw_cephadm_dashboard_key | default("") | length > 0
```

fmount reviewed (approved):
/lgtm

pablintino commented:
/approve

fultonj commented:
Most recent change fixed it in my env.

```yaml
included: /home/zuul/src/github.com/openstack-k8s-operators/ci-framework/roles/cifmw_cephadm/tasks/dashboard/validation.yml for compute-xxusapf1-0

TASK [cifmw_cephadm : Update urischeme based on cert/key cifmw_cephadm_urischeme_dashboard=https] ***
Friday 27 September 2024  10:16:00 -0400 (0:00:00.063)       0:02:23.815 ****** 
ok: [compute-xxusapf1-0]

TASK [cifmw_cephadm : Validate connection to dashboard service url={{ cifmw_cephadm_urischeme_dashboard | default('http') }}://{{ grafana_server_addr }}:{{ cifmw_cephadm_dashboard_port }}, dest=/tmp/dash_response, validate_certs=False] ***
Friday 27 September 2024  10:16:00 -0400 (0:00:00.065)       0:02:23.881 ****** 
changed: [compute-xxusapf1-0]

TASK [cifmw_cephadm : Check http response code from dashboard service with login url={{ cifmw_cephadm_urischeme_dashboard | default('http') }}://{{ grafana_server_addr }}:{{ cifmw_cephadm_dashboard_port }}, dest=/tmp/dash_http_response, validate_certs=False, username=admin, password=admin] ***
Friday 27 September 2024  10:16:01 -0400 (0:00:00.682)       0:02:24.563 ****** 
changed: [compute-xxusapf1-0]
```

fmount reviewed (approved):
/lgtm

fultonj commented:
/cherrypick 18.0.0-proposed

fultonj commented:
cherry picked here https://github.com/openstack-k8s-operators/ci-framework/pull/2405
